lib/player.rb

```diff
@@ -12,44 +12,116 @@ def add_ship(ship)
     @ships << ship
   end
 
+
+
   def ship_setup
     if @type == :auto
-      # helper-method: auto_generation CALLS
-      # helper-method: player_place_ships
-      # auto generation
-      p "type auto"
-    elsif @type == :user
-      # Put in a guard for user error
-
-      # helper-method: player_place_ships
-      @ships.each do |ship|
-        if ship.length == 3
-          puts "Enter the squares for the Cruiser (3 spaces):"
-          puts "> "
-        elsif ship.length == 2
-          puts "Enter the squares for the Submarine (2 spaces):"
+      until all_ships_placed?
+        auto_generate_coordinates
+      end
+    else
+      until all_ships_placed?
+        @ships.each do |ship|
+          puts "Enter the squares for the #{ship.name} (#{ship.length} spaces)"
```

Comment:
Love how you made this dynamic! Way to think ahead for iteration 4. If we get time to make the board more dynamic for user input, I super look forward to refactoring and updating all this code (really, just everything for iteration 4 is exciting!)